src/areq/exceptions.py

```
@@ -0,0 +1,285 @@
+        # its requests-side parents. Our AreqException is a parallel base.
+        # The key is that AreqException sets up self.underlying_exception and self.request.
+        # And the final exception type is requests.exceptions.ConnectTimeout.
+
+        # Alternative: Let super() handle it if MRO is set up for AreqException first.
+        # super(AreqConnectTimeout, self).__init__(error)
+        # This would call AreqTimeout.__init__ -> AreqException.__init__
+        # Then requests.ConnectTimeout needs to be initialized. This is tricky.
+
+        # Simplest for multiple inheritance with a common custom base:
+        # Initialize the custom base directly, then the primary library base.
+        # AreqException.__init__(self, error) # Already done by AreqTimeout or AreqConnectionError if we used super()
```

Pylint (3.3.7)

[convention] Line too long (117/100)

(C0301)